src/eos-metrics-instrumentation.c

```
@@ -75,6 +86,94 @@ static gint64 start_time;
 
+static gboolean
+get_os_version (gchar **name_out,
+                gchar **version_out)
+{
+    g_autoptr(GError) error = NULL;
+    gboolean ret = FALSE;
+
+    g_autoptr(GFile) os_release_file = g_file_new_for_path (OS_RELEASE_FILE);
+    g_autoptr(GFileInputStream) file_stream = g_file_read (os_release_file, NULL, &error);
+    if (error)
+      goto out;
+
+    g_autoptr(GDataInputStream) data_stream = g_data_input_stream_new (G_INPUT_STREAM (file_stream));
+
+    g_autofree gchar *name = NULL;
+    g_autofree gchar *version = NULL;
+    gchar *line;
+    while ((!name || !version) &&
+           (line = g_data_input_stream_read_line (data_stream, NULL, NULL, &error)))
+      {
+        if (g_str_has_prefix (line, "NAME="))
+          name = line;
+        else if (g_str_has_prefix (line, "VERSION="))
+          version = line;
+        else
+          g_free (line);
+      }
+
+    if (error)
+      goto out;
+    if (!name || !version)
+      {
+        g_warning ("Could not find NAME or VERSION keys in " OS_RELEASE_FILE ".");
+        goto out;
+      }
+
+    /* According to os-release(5), these values can be quoted, escaped,
+     * etc. For simplicity, instead of doing the parsing on the client
+     * side, we do it on the server side.
+     */
+    *name_out = g_strdup (name + strlen ("NAME="));
+    *version_out = g_strdup (version + strlen ("VERSION="));
+
+    ret = TRUE;
+
+ out:
+    if (error)
+      g_warning ("Error reading " OS_RELEASE_FILE ": %s.", error->message);
+
+    return ret;
+}
+
+static void
+get_eos_personality (gchar **personality_out)
+{
+    gchar *personality = NULL;
+    g_autoptr(GKeyFile) key_file = g_key_file_new ();
+
+    if (!g_key_file_load_from_file (key_file, PERSONALITY_FILE_PATH,
+                                    G_KEY_FILE_NONE, NULL))
+      goto out;
+
+    personality = g_key_file_get_string (key_file, PERSONALITY_CONFIG_GROUP,
+                                         PERSONALITY_KEY, NULL);
```

Should we silently ignore this error?